### ISSUE TITLE
Add some propagation specific training visualization

### ISSUE THREAD

**Issue by seemasingh** [encog/encog-java-core#58]
It would be useful to allow certain aspects of training to be visualized.  For example show the gradients and deltas in RPROP.  This could help to identify causes of training failure, such as local minima, overfitting and other issues.
